File: 프로그래머스/lv2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.js
function solution(elements) {    
    const numSet = new Set();
    const n = elements.length;
    
    for (let i = 0; i < n; i++) {
        let result = 0;
        for (let j = i; j < i + n; j++) {
            result += elements[j % n];
            numSet.add(result)
        }
    }
    

    return numSet.size;
}

File: 프로그래머스/lv2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.py
def solution(elements):
    n = len(elements)
    num_list = set()
    
    for i in range(0, n):
        result = 0
        for j in range(i, i + n):
            result += elements[j % n]
            num_list.add(result)
    
    return len(num_list)